pyrefly/lib/state/lsp.rs

```diff
@@ -942,6 +984,46 @@ impl<'a> Transaction<'a> {
         }
     }
 
+    fn active_parameter_index(params: &[Param], active_argument: &ActiveArgument) -> Option<usize> {
+        match active_argument {
+            ActiveArgument::Positional(index) | ActiveArgument::Next(index) => {
+                (*index < params.len()).then_some(*index)
+            }
+            ActiveArgument::Keyword(name) => params.iter().position(|param| {
+                Self::parameter_name(param).is_some_and(|param_name| param_name == name)
+            }),
+        }
+    }
+
+    fn parameter_name(param: &Param) -> Option<&Name> {
+        match param {
+            Param::PosOnly(Some(name), ..)
+            | Param::Pos(name, ..)
+            | Param::VarArg(Some(name), ..)
+            | Param::KwOnly(name, ..)
+            | Param::Kwargs(Some(name), ..) => Some(name),
+            _ => None,
```

Comment:
looks like this covers all cases so please exclude this wildcard (makes it less bug-prone when we add a new Param)